src/itensor.jl

```diff
@@ -2565,8 +2565,20 @@ function HDF5.read(
   # TODO: use Vector{Index} here?
   inds = read(g, "inds", IndexSet)
 
-  stypestr = read(attributes(open_group(g, "storage"))["type"])
-  stype = eval(Meta.parse(stypestr))
-  storage = read(g, "storage", stype)
-  return itensor(storage, inds)
+  # check input file for key name of ITensor data
+  # ITensors.jl <= v0.1.x uses `store` as key 
+  # whereas ITensors.jl >= v0.2.x uses `storage` as key
+  if haskey(g, "storage")
```

Comment:
Maybe compress to something like:
```julia
for k in ["store", "storage"]
  if haskey(g, k)
    stypestr = read(attributes(open_group(g, k))["type"])
    stype = eval(Meta.parse(stypestr))
    storage = read(g, k, stype)
    return itensor(storage, inds)
  end
end
```

Reply:
Okay, that makes sense!
Would you like to quietly leave the if condition without an error in a case (albeit a pathological one) where a stored ITensor does neither have key `storage` nor `store`?

Reply:
Your above code does however render the function to have a predictable type as output...

Reply:
You could keep the same error message you had and put it after the for loop.